File: atcoder/arc/arc062_a.py
N = int(input())
T = A = 1
for _ in range(N):
    t, a = map(int, input().split())
    k = max([1, T // t + (T % t != 0), A // a + (A % a != 0)])
    T, A = k * t, k * a
print(T + A)

File: atcoder/arc/arc062_a.nim
import strutils, sequtils

let N = stdin.readline.parseInt
var T, A = 1.int64

for _ in 0..<N:
  let
    input = stdin.readline.split.map(parseBiggestInt)
    (t, a) = (input[0], input[1])
    k = max(1.int64, max(T div t + (T mod t != 0).int64, A div a + (A mod a != 0).int64))
  (T, A) = (k * t, k * a)
echo(T + A)
